server/drivers/hd44780.c

```diff
@@ -238,30 +243,62 @@ static int parse_backlight_option( const char *value )
 	return -1;
 }
 
+static void strappend(char *dst, size_t dsize, const char *src) {
+
+	size_t dlen = strlen(dst);
+	size_t slen = strlen(src);
+
+	if (slen + dlen < dsize) {
+		memcpy(dst + dlen, src, slen);
+		dst[dlen+slen] = 0;
+	}
+	else if (dlen < dsize) {
+		memcpy(dst + dlen, src, slen - (dsize - dlen));
```

Comment:
I think here you just want `dsize - dlen` instead of `slen - (dsize - dlen)` ...

Reply:
Yes, you are right

Reply:
I'm not saying that manually parsing true/false values is better, but it costs nothing in this case, as  that complication of function `parse_backlight_option()` is because of handling list of options and assumption that only few of them can be combined (first three) and `none` shall be single one (as this looks natural). 

It would be only little simpler if it would be allowed to add `none` to list, as it is harmless (value is 0).

Reply:
It's just adds few records to table `bl_value_mapping` with boolean aliases.

On the other hand I think as there is specialized function for parsing value, it should handle all possible values, even if it doesn't look simple.

Reply:
> It's just adds few records to table bl_value_mapping with boolean aliases.

Yes, having redundant definition of the boolean values is one of the downsides IMO.

> On the other hand I think as there is specialized function for parsing value, it should handle all possible values, even if it doesn't look simple and don't do the same in other place.

Here I fail to see the rationale. Can you explain this to me?

As I see it the boolean values are basically deprecated and only supported for backwards compatibility. I think it is straigt forward to have one function parsing the real thing and the backwards compatibility fallback at the call site.

I don't have a strong opinion on this, but I really would like to understand the merit you see before merging something that feels counter-intuitive and complicated to me.